camel/storages/vectordb_storages/milvusDB_v2.py

```diff
@@ -0,0 +1,99 @@
+from pymilvus import MilvusClient, Collection
+from typing import Any, Dict, List, Optional, Tuple, Union, cast
+import uuid
+
+from camel.storages.vectordb_storages import (
+    BaseVectorStorage,
+    VectorDBQuery,
+    VectorDBQueryResult,
+    VectorDBStatus,
+    VectorRecord,
+)
+
+class MilvusStorage(BaseVectorStorage):
+    """
+    This module is to integration of Milvus, a vector database, into Camel agent.
+    The information of Milvus can be accessed to Milvus websites at: https://milvus.io/docs
+
+    Args:
+        vector_dim (int): The dimenstion of storing vectors.
+        collection_name (Optional[str], optional): Name for the collection in
+            the Milvus. If not provided, set it to the current time with iso
+            format. (default: :obj:`None`)
+        url_and_api_key (Optional[Tuple[str, str]], optional): Tuple containing
+            the URL and API key for connecting to a remote Milvus instance.
+            (default: :obj:`None`)
+        path (Optional[str], optional): Path to a directory for initializing a
+            local Milvus client. (default: :obj:`None`)
+        distance (VectorDistance, optional): The distance metric for vector
+            comparison (default: :obj:`VectorDistance.COSINE`)
+        **kwargs (Any): Additional keyword arguments for initializing
+            `MilvusClient`.
+    """
+
+    # Initialize the MilvusStorage class with required parameters
+    def __init__(self, vector_dim: int, url_and_api: Tuple[str, str], **kwargs: Any) -> None:
+        self._client: MilvusClient
+        self.vector_dim = vector_dim
+        self.create_client(url_and_api, **kwargs)   
+        self.collection_name = self.create_collection()
+
+    # Create a Milvus client using the provided URL and API key
+    def create_client(self, url_and_api_key: Tuple[str, str], **kwargs: Any) -> None:
+        self._client = MilvusClient(uri=url_and_api_key[0], token=url_and_api_key[1], **kwargs)
+
+    # Create a new collection in Milvus with a unique name and vector dimensions
+    def create_collection(self, **kwargs: Any,) -> str:
+        try:    
+            collection_name  = self.generate_collection_name()
+            self._client.create_collection(collection_name = collection_name, dimension = self.vector_dim, **kwargs,)
```

Comment:
need `index_params` define metric_type and index_type,  `schema`  define metadata when creating collection

Reply:
I updated this two parameters (index_params and schema) in my new version